diesel/src/pg/types/ranges.rs

```diff
@@ -0,0 +1,125 @@
+
+use byteorder::{ReadBytesExt, WriteBytesExt, NetworkEndian};
+use backend::Debug;
+use pg::{Pg, PgTypeMetadata};
+use types::*;
+use std::collections::Bound;
+use std::error::Error;
+use query_source::Queryable;
+
+bitflags! {
+    struct RangeBound: u8 {
+        const RANGE_EMPTY = 0x01;
+        const RANGE_LB_INC = 0x02;
+        const RANGE_UB_INC = 0x04;
+        const RANGE_LB_INF = 0x08;
+        const RANGE_UB_INF = 0x10;
+        const RANGE_LB_NULL = 0x20;
+        const RANGE_UB_NULL = 0x40;
+        const RANGE_CONTAIN_EMPTY = 0x80;
+    }
+}
+
+#[derive(Clone, Copy, Debug)]
+pub struct RangeMetadata {
+    oid: u32,
+    array_oid: u32
+}
+
+pub trait CanAppearInRange {
+    fn metadata() -> RangeMetadata;
```

Comment:
Hm... This feels off to me. I think I'd prefer this to be something like:

```rust
pub trait CanAppearInRange where
    Pg: HasSqlType<Self::RangeType>,
{
    type RangeType;
}
```

That way we can just call `metadata` on `RangeType`